File: person.js
class Person {
    constructor(name = "Hopago", Strong = 100, Smart = 100) {
        this.name = name;
        this.Strong = Strong;
        this.Smart = Smart;
    }

    static guDoc = "Dopago"

    static talk(a, b) {
        if ((a && b) instanceof Person) {
            console.log(`${a.name}과 ${b.name}의 대화...`)
        }
    }

    levelUp() {
        this.Strong++;
        this.Smart++;
        console.log(`${this.name}의 통찰력: ${this.Smart}, 힘: ${this.Smart}`)
    }
}

const aPerson = new Person();
const bPerson = new Person("도파고", 1, 1000);

aPerson.levelUp();
bPerson.levelUp();
Person.talk(aPerson, bPerson);
console.log(Person.guDoc);

export {
    Person,
    aPerson,
    bPerson,
}

File: Person.java
public class Person {
    String name;
    int Strong;
    int Smart;

    public Person(String name, int Strong, int Smart) {
        this.name = name;
        this.Smart = Smart;
        this.Strong = Strong;
    }

    public Person() {
        this.name = "Hopago";
        this.Smart = 100;
        this.Strong = 100;
    }

    public void levelUp() {
        this.Strong = this.Strong + 1;
        this.Smart = this.Smart + 1;
        System.out.println(this.name + "의 통찰력과 힘, 증가");
    }

    private static void talk(Person a, Person b) {
        System.out.println(a.name + " & " + b.name + "의 대화 시작...");
    }

    public static void main(String[] args) {
        Person a = new Person();
        a.levelUp();

        Person b = new Person("도파고", 100, 100);
        b.levelUp();

        Person.talk(a, b);
    }
}
